### PR TITLE
[BEAM-1348] Remove deprecated concepts in Fn API (now replaced with Runner API concepts).

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @robertwb 
CC: @kennknowles

File: sdks/common/fn-api/src/main/proto/beam_fn_api.proto [hunk 1]
```diff
@@ -278,33 +154,20 @@ message ProcessBundleDescriptor {
...
-  repeated PrimitiveTransform primitive_transform = 2 [deprecated = true];
-
-  // (Deprecated) The set of all coders referenced in this bundle.
-  //
-  // Migrate to Runner API defintions found within codersyyy field.
-  repeated Coder coders = 4 [deprecated = true];
-
   // (Required) A map from pipeline-scoped id to PTransform.
-  map<string, org.apache.beam.runner_api.v1.PTransform> transforms = 5;
+  map<string, org.apache.beam.runner_api.v1.PTransform> transforms = 2;
```
Comment (robertwb) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Could we be re-using the runner API Components message here?

Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: To my knowledge, Components is meant as a short term message to pass data through Dataflow's service while the model transitions, its not expected to be used outside of Dataflow and should be deleted once Dataflow can transition off of it.

Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Spoke with @kennknowles, he says MessageWithComponents is the Dataflow specific thing. I am currently discussing with him whether we should take on Components here or not.

In the mean time, this is still forward progression whether we adopt Components here or not. Do you think we can merge now and address migration to Components when that discussion comes to a conclusion.

Comment (robertwb) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: MessageWithComponents is a dataflow-specific hack. Components (perhaps renamed) seems more general. LGTM on this PR now though.

Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Great.

Comment (kennknowles) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: Let's not be too mean to `MessageWithComponents`. It is easy to imagine other runners wanting to package up a tiny piece of a pipeline. We have far greater hacks in our codebase - this one you could consider a general utility :-)

Comment (lukecwik) [inline sdks/common/fn-api/src/main/proto/beam_fn_api.proto]: If runners need to do something like this, they should hide it deep within their bowels and not pollute the public portion.

**Pull request closed without merging**